# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentLoadListener.h
# Commit: bfe21f274ed5
# Full Hash: bfe21f274ed5a4630a3503bcb875fed04f90533c
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-16 03:00:04
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 29ff609063c2 -r bfe21f274ed5 netwerk/ipc/DocumentLoadListener.h
--- a/netwerk/ipc/DocumentLoadListener.h	Wed Apr 15 18:53:22 2020 +0000
+++ b/netwerk/ipc/DocumentLoadListener.h	Wed Apr 15 18:53:06 2020 +0000
@@ -96,11 +96,11 @@
                                 ADocumentChannelBridge* aBridge);
 
   // Creates the channel, and then calls AsyncOpen on it.
-  bool Open(nsDocShellLoadState* aLoadState, LoadInfo* aLoadInfo,
-            nsLoadFlags aLoadFlags, uint32_t aCacheKey,
-            const uint64_t& aChannelId, const TimeStamp& aAsyncOpenTime,
-            nsDOMNavigationTiming* aTiming, Maybe<dom::ClientInfo>&& aInfo,
-            uint64_t aOuterWindowId, bool aHasGesture, nsresult* aRv);
+  bool Open(nsDocShellLoadState* aLoadState, nsLoadFlags aLoadFlags,
+            uint32_t aCacheKey, const uint64_t& aChannelId,
+            const TimeStamp& aAsyncOpenTime, nsDOMNavigationTiming* aTiming,
+            Maybe<dom::ClientInfo>&& aInfo, uint64_t aOuterWindowId,
+            bool aHasGesture, nsresult* aRv);
 
   NS_DECL_ISUPPORTS
   NS_DECL_NSIREQUESTOBSERVER
@@ -214,8 +214,6 @@
                         nsTArray<ParentEndpoint>&& aStreamFilterEndpoints);
 
   // Construct a LoadInfo object to use for the internal channel.
-  // TODO: This currently only supports creating top window TYPE_DOCUMENT
-  // LoadInfos
   already_AddRefed<LoadInfo> CreateLoadInfo(
       dom::CanonicalBrowsingContext* aBrowsingContext,
       nsDocShellLoadState* aLoadState, uint64_t aOuterWindowId);